src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineReceivedStamp.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Symfony\Component\Messenger\Stamp\StampInterface;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class DoctrineReceivedStamp implements StampInterface
+{
+    private $doctrineEnvelope;
+
+    /**
+     * @param $doctrineEnvelope
+     */
+    public function __construct(array $doctrineEnvelope)
```

Comment:
You don't need all this "doctrine envelope". For now, you just need the `id` actually, let's just have this for now. Otherwise, we will apply the BC to this entire envelope, it's a bit too much. Later we might add the `available_at`, etc... but I don't believe this would ever be needed.